### DISCUSSION TITLE
0603 피어세션일지

### DISCUSSION THREAD

**Discussion by gvsteve24** [bcaitech1/p4-dkt-freshtomato#69] [General]
# 오늘의 TMI + 이모지

* [진현] 운영진님과 dm 주고받음 ㅋㅋ 퇴실을 늦게 찍어서!
* [정현] 잠깐 고개를 젖혔는데 그대로 잠이 듬

| 진현       | 정현               | 유라     | 아라       | 재훈   | 민용       |
| ---------- | ------------------ | -------- | ---------- | ------ | ---------- |
| :umbrella: | :man_technologist: | :burger: | :magician: | :hole: | :question: |

* 가고 싶은 회사

| 진현       | 정현 | 유라                | 아라       | 재훈       | 민용       |
| ---------- | ---- | ------------------- | ---------- | ---------- | ---------- |
| :question: | Riiid | 네이버 웹툰 / Clova | 네이버 Clova  | :question: | :question: |

* 최애 웹툰 

| 진현 | 정현          | 유라            | 아라 | 재훈          | 민용       |
| ---- | ------------- | --------------- | ---- | ------------- | ---------- |
| 내일 | 유미의 세포들 | 선천적 얼간이들 | 바보 | 죽음에 관하여 | :question: |

# 회의 및 토론

### 현재 실험 상황

[아라__SAINT_]  실험 진행중

* baseline기준. 피처 세개

* 중간스코어

| LB 미제출 | SAINT  | LSTM   |
| --------- | ------ | ------ |
| Valid_AUC | 0.76xx | 0.77xx |

* 학습 소요시간: 3 hrs
  10 fold 기준 , layer 개수 , max_seq를 늘렸어서 오래 걸림
  기존 세팅 이랑 비슷하게 하면 에폭당 15분
* 향후 계획
  정오답 개수를 continuous로 바꿔서 실험 해보려고 함 
  LSTM GRU BERT 는 그대로 계속 할 듯

--

[진현__논문_] 진행중

* input 어떻게 되는지까지

* 논문에서 나온 인풋 및 피처를 따라가서 해보면 현재결과가 개선될거 같다.

![image-20210603170202698](C:\Users\JUNGHYUN LEE\AppData\Roaming\Typora\typora-user-images\image-20210603170202698.png)

**논문에서 나온 하이퍼파라미터 및 세팅**

| Type           |       |
| -------------- | ----- |
| Optimizer      | Adam  |
| Window size    | 100   |
| Batch size     | 128   |
| Xavier Uniform |       |
| Dropout rate   | 0.1   |
| warmup_steps   | 4000  |
| learning rate  | 0.001 |

**SAINT 관련 이슈 토의**

* 현재 시도들

문제관련 피처 3개 encoder 들어가고 / encoder들어간 것들 + interaction(응답) 추가한게 decoder로 들어가는 상황 

* interaction만 decoder에 넣어보자 
  * 결과) [민용] 떨어짐
    		  [아라] 비슷

* (Testid , assessmentitemid)는 인코더에 Interaction은 디코더에 넣고 해볼까!

* 노트북으로 제공된 것중에 Last Query도 있으니 실험해볼 예정

* Saint의 exerciseId  랑 iscream data assessmentItemID 의미 같은 건지 차이 있는지

[재훈__feature selection_] 진행중

* 결측치 추가 관련: na를 채웠을 때 좀 떨어진다. 

* lightgbm 결측치를 채울때 전체 데이터의 평균을 취해서 넣어줬는데 떨어지는 모습

![image-20210603173011366](C:\Users\JUNGHYUN LEE\AppData\Roaming\Typora\typora-user-images\image-20210603173011366.png)

* [결론] 노트북으로 공유하면서 valid셋 나누는걸 좀 확실히 해봐야겠다.

* Catboost관련해서 모델이름 넣고 바로 돌리게 노트북 업데이트함

**[:question:]** **Riiid 대회 데이터 활용하기** 

연관성있는 컬럼을 찾아서 데이터를 탐색하기

베이스라인에 RIIID데이터를 통과를 해서 먼저 valid 추이를 보기.

이어서 Finetuning을 하면서 validation score추이를 보기

노트북을 가지고 합해진 데이터를 가지고 마지막문제만 -1 처리를 하고 

LSTM에 돌아가는지만 확인해보기!

* testId, KnowledgeTag같은 경우가 있어서 데이터를 트리계열에 쓰는 것은 안될거같음

**Feature 추가한거랑 notebook등 업데이트해서 PR, Merge** 

* ~~SainT구조까지만 잡고~~ 머지할 예정 (SAINT 는 일단 보류)
* continuous한 피처를 적어도 하나 넣어줘야될거같아서 반영할 예정

# 오늘의 저녁메뉴

| 진현       | 정현 | 유라 | 아라   | 재훈   | 민용   |
| ---------- | ---- | ---- | ------ | ------ | ------ |
| 샐러드 | 김밥 | 빵   | 햄버거 | 불고기 | 떡갈비 |

**Comment by arabae** [comment]
- [x] [실험] Last Query model 구조 확인
- [x] [실험] Last Query model에 사용된 feature 확인
- [x] [실험] Last Query model 실험
- [ ] [논문] SAINT 논문 읽기 마무리
- [x] [네트워킹] 이력서 완성

**Comment by gvsteve24** [comment]
**- [ ] Todo List**
- [x] [실험] Riid데이터컬럼별 통계량 확인
- [ ] [실험] 활용할 수 있는 컬럼(testid, itemid, tag, answer)으로 변환하기
- [ ] [실험] Train/Valid 셋 나누고 모델에 돌려보기
- [ ] [논문] DKVMN model ppt에 정리
- [x] [네트워킹] 이력서 작성 완료
